server/src/session_aware_backend_handler.rs

```diff
@@ -0,0 +1,441 @@
+        let handler = SessionAwareBackendHandler::new(mock_sql_backend, jwt_blacklist.clone());
+
+        let user_id = UserId::new("test_user");
+        let request = UpdateUserRequest {
+            user_id: user_id.clone(),
+            login_enabled: Some(true),
+            ..Default::default()
+        };
+
+        // This test verifies that enabling login doesn't trigger session invalidation
+        // Since we're using an actual SqlBackendHandler with an in-memory database,
+        // the update_user call may fail due to the user not existing, but that's expected
+        // The key is that no session invalidation logic should be triggered
+        let initial_blacklist_size = jwt_blacklist.read().unwrap().len();
+
+        // We expect this to fail because the user doesn't exist in the database
+        let _result = handler.update_user(request).await;
+
+        // Verify that no tokens were added to the blacklist
+        assert_eq!(jwt_blacklist.read().unwrap().len(), initial_blacklist_size);
+    }
+
+    #[tokio::test]
+    async fn test_update_user_with_no_login_enabled_field() {
+        let mock_sql_backend = create_test_backend().await;
+        let jwt_blacklist = Arc::new(RwLock::new(HashSet::new()));
+        let handler = SessionAwareBackendHandler::new(mock_sql_backend, jwt_blacklist.clone());
+
+        let user_id = UserId::new("test_user");
+        let request = UpdateUserRequest {
+            user_id: user_id.clone(),
+            login_enabled: None, // No login_enabled field change
+            ..Default::default()
+        };
+
+        let initial_blacklist_size = jwt_blacklist.read().unwrap().len();
+
+        // We expect this to fail because the user doesn't exist in the database
+        let _result = handler.update_user(request).await;
+
+        // Verify that no tokens were added to the blacklist
+        assert_eq!(jwt_blacklist.read().unwrap().len(), initial_blacklist_size);
+    }
+
+    #[tokio::test]
+    async fn test_inner_accessor() {
+        let mock_sql_backend = create_test_backend().await;
+        let jwt_blacklist = Arc::new(RwLock::new(HashSet::new()));
+        let handler = SessionAwareBackendHandler::new(mock_sql_backend, jwt_blacklist);
+
+        // Test that we can access the inner SqlBackendHandler
+        let _inner = handler.inner();
+    }
+
+    #[tokio::test]
+    async fn test_jwt_blacklist_shared_reference() {
+        let mock_sql_backend = create_test_backend().await;
+        let jwt_blacklist = Arc::new(RwLock::new(HashSet::new()));
+
+        // Add a token to the blacklist before creating the handler
+        jwt_blacklist.write().unwrap().insert(12345);
+
+        let _handler = SessionAwareBackendHandler::new(mock_sql_backend, jwt_blacklist.clone());
+
+        // Verify that the handler shares the same blacklist reference
+        assert_eq!(jwt_blacklist.read().unwrap().len(), 1);
+        assert!(jwt_blacklist.read().unwrap().contains(&12345));
+    }
```

Comment:
erm, you're checking that `clone()` does a shallow clone? No need, that's part of the interface. Remove the test.